client/js/restaurants_api.js

```diff
@@ -679,6 +662,16 @@ function createNewReservation(restaurant_name) {
         'startTime': reservation_datetime
       };
 
+  var data = {
+    '@type': 'FoodEstablishmentReservation',
+    'partySize': partySize,
+    'reservationFor': {
+      '@type': 'FoodEstablishment',
+      'name': '' + restaurant_name
+    },
+    'startTime': reservation_datetime+'Z'
```

Comment:
We need to use a date and time so,  datetime object is needed. Currently, datetime object is not supported by HTML5 and datetime-local has a lack of UI implementation. For instance https://jsfiddle.net/yvx2vv3z/ has a nice input method in Chrome but none in Safari or Firefox.

We thought to use two different fields for date and time and use custom pickers. Those pickers allow us to retrieve the values with the desired string formats. We use strings because the API expects a valid string as date ( http://docs.tourguide.apiary.io/#reference/api-especification-for-tourguide/reservation/create-reservation). We think this approach could be more efficiently than create a date object and convert it to string.

The ```+Z``` code is used to pass the server side filter that does not accept dates without specify the milliseconds.